accepted/00101-native-appstreams.md

```diff
@@ -0,0 +1,75 @@
+
+The goal is to seamlessly support modularity in its native form within uyuni, ensuring a user-friendly and intuitive experience across all workflows involving modular repositories.
+
+# Detailed design
+[design]: #detailed-design
+
+## Overview
+
+The design encompasses core functionality, new UI operations, modifications to the existing CLM feature, and testing infrastructure.
+
+## The core design
+
+The core design involves three key components for data collection and filtering of packages for client consuption:
+
+1. **Repo-sync**: A standalone Python module that extracts module metadata from source channels using `dnf`'s `libmodulemd` library during repo-sync, serving as the source of truth for the module information. As the data gathered during this process is critical to the feature, it must be executed as a single unit with the repo-sync ([See the PoC](https://github.com/cbbayburt/uyuni/commit/ed9391e8c6e0a66d1dd7cb0f3501332b0884f2f3)).
+
+2. **Package profile update**: A Salt state that retrieves current module information from clients, storing the data in the database. The state calls `dnf module` commands to retrieve the list of enabled module streams together with their *name*, *stream*, *version*, *context*, and *arch* values ([see the PoC](https://github.com/cbbayburt/uyuni/commit/2c788f3144f5bfe8ddd904045e0a757a7a432923)).
+
+3. **Package cache**: Incorporates data from repo-sync and package profile update into existing queries, ensuring proper filtering of non-enabled modular packages. The existing tables, queries and stored procedures will be updated with additional joins and proper indexing to ensure minimum performance impact.
+
+A prototype showcasing the core functionality can be found [here](https://github.com/cbbayburt/uyuni/tree/native_appstreams_poc).
+
+### ER diagram
+
+The following figure shows the proposed ER diagram for the core design.
+
+![ER diagram](images/00101-native-appstreams-er-diagram.png)
+
+## New UI operations
+
+The proposed UI includes a new page under the `System > Software` tab, visible only for clients with modular channels assigned. It displays available modules and their streams, allowing users to select and apply module changes to clients.
+
+The main workflow is as follows:
+
+1. Display modules and their available streams from all the assigned modular channels
+2. The user selects module streams to be enabled on the client
+3. *Optional:* Run `mgr-libmod` for dependency resolution with real-time feedback.
```

Comment:
Again, I lack knowledge here so another question. What kind of dependency issue would you expect? Do streams depend on other streams or are they self-contained?

Ideally, we should have mechanism in place, so user doesn't get into bad state. Just like we do with required/recommended channels. Pre-select the needed all the needed stuff when user select a module stream. But if that's asking for too much, this optional would be super helpful.

Reply:
Yes, a module may depend on other modules and they must be enabled together. If we miss this, dnf would complain about the missing dependencies. Other than that, mgr-libmod also checks for conflicting streams. If, for example the user tries to enable two streams of the same module, mgr-libmod would complain about this too.